Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/AnimGraph/ContextMenu.cpp

```diff
@@ -43,18 +43,37 @@ namespace EMStudio
 
         QMenu* nodeGroupMenu = new QMenu(tr("Assign To Node Group"), menu);
 
+        // We keep track of how many groups are valid for the level currently being shown on the animation graph.
+        // If there is none, we won't show the "assign" action on the context menu.
+        int validGroupCount = 0;
+
         for (size_t i = 0; i < numNodeGroups; ++i)
         {
             EMotionFX::AnimGraphNodeGroup* nodeGroup = animGraph->GetNodeGroup(i);
+            auto* activeGraph = GetActiveGraph();
+
+            auto nodeGroupParentId = nodeGroup->ParentNodeId();
+            auto currentLevelParentIndex = activeGraph->GetAnimGraphModel().GetFocus();
+            auto currentLevelParentId = currentLevelParentIndex.isValid()
+                ? currentLevelParentIndex.data(AnimGraphModel::ROLE_ID).value<EMotionFX::AnimGraphNodeId>()
+                : EMotionFX::AnimGraphNodeId{};
```

Comment:
Seems like the following code can be moved outside the loop and therefore only done once.

````
            auto* activeGraph = GetActiveGraph();
            auto currentLevelParentIndex = activeGraph->GetAnimGraphModel().GetFocus();
            auto currentLevelParentId = currentLevelParentIndex.isValid()
                ? currentLevelParentIndex.data(AnimGraphModel::ROLE_ID).value<EMotionFX::AnimGraphNodeId>()
                : EMotionFX::AnimGraphNodeId{};
````

Reply:
Not relevant here, I mistakenly force pushed changes from a different PR here.